bookdown/03-optimization-searchspace.Rmd

```diff
@@ -0,0 +1,223 @@
+## Tuning Search Spaces {#searchspace}
+
+When running an optimization, it is important to inform the tuning algorithm about what hyperparameters are valid. Here the names, types, and valid ranges of each hyperparameter are important. All this information is communicated with objects of the class `r ref("ParamSet")`, which is defined in `r mlr_pkg("paradox")`. While it is possible to create `r ref("ParamSet")`-objects using its `$new`-constructor, it is much shorter and readable to use the `r ref("ps")`-shortcut, which will be presented here. For an in-depth description of `r mlr_pkg("paradox")` and its classes, see the [`paradox` chapter](#paradox).
+
+An important distinction to keep in mind when defining `r ref("ParamSet")` objects is whether these objects are used to define a search space for tuning, or whether they are only used to check the validity of values. The `minsplit` parameter of the `r ref("mlr_learners_classif.rpart", "classif.rpart Learner")`, for example, has a lower but no upper bound. When tuning the value, however, a lower *and* upper bound must be given because tuning search spaces need to have finite measure. `r ref("ParamSet")` objects are used to describe both the search spaces of tuning runs, and the valid parameter ranges of `Learner`s and other objects. Therefore it is possible to create "unbounded" `r ref("ParamSet", text = "ParamSets")`, mostly in advanced usecases like creating or modifying `r ref("Learner")` or `r ref("PipeOp")` objects. Here, however, we will mainly focus on creating "bounded" `r ref("ParamSet", text = "ParamSets")` that can be used for tuning. See the [in-depth `paradox` chapter](#paradox) for more details on using `r ref("ParamSet", text = "ParamSets")` to  valid hyperparameter ranges for use-cases besides tuning.
```

Comment:
```suggestion
Note, that `r ref("ParamSet")` objects exist in two contexts.
First, `r ref("ParamSet")`-objects are used to define the space of valid parameter setting for a learner (and other objects).
Second, they are used to define a search space for tuning.
We are mainly interested in the latter.
For an example we can consider the `minsplit` parameter of the `r ref("mlr_learners_classif.rpart", "classif.rpart Learner").
The `r ref("ParamSet")` associated with the learner has a lower but *no* upper bound.
However, for tuning the value, a lower *and* upper bound must be given because tuning search spaces need to be bounded. 
For `r ref("Learner")` or `r ref("PipeOp")` objects, typically "unbounded" `r ref("ParamSet", text = "ParamSets")` are used.
Here, however, we will mainly focus on creating "bounded" `r ref("ParamSet", text = "ParamSets")` that can be used for tuning. 
See the [in-depth `paradox` chapter](#paradox) for more details on using `r ref("ParamSet", text = "ParamSets")` to  define parameter ranges for use-cases besides tuning.
```
one line per sentence would help here :)